### ISSUE TITLE
static and non-static methods

### ISSUE THREAD

**Issue by Ustinian-yu-code** [Ustinian-yu-code/python-questions#5]
Define the difference between static and non-static methods that function calculates the distance between two points

**Comment by yeziarrietty** [comment]
Non-static methods require an instance of the class to be created, and then the method is called from the instance. It is closely related to instances of the class, each with its own independent copy of the method. This makes it suitable for performing actions in the context of an object instance.
Static methods are defined by @staticmethod decorators, which do not require a self argument because it does not depend on an instance of the class. It is suitable for implementing some general-purpose functions that do not depend on the state of a particular object, and perform purely mathematical calculations.
In general, non-static methods are closely related to instances of classes, with the ability to access and manipulate the properties of instances; Static methods, on the other hand, are more like class-related utility functions that do not depend on the instance state, providing a more flexible way to invoke the relevant functions without the need for an instance.
